src/lib/helpers.ts

```diff
@@ -220,6 +220,53 @@ export function getOvalConfigs(input: string): OvalConfigs {
   throw new Error(`Value "${input}" is valid JSON but is not OvalConfigs records`);
 }
 
+// Get OvalHeaderConfigs from the header string or throw an error if the header is invalid.
+export const getOvalHeaderConfigs = (
+  header: string | string[] | undefined,
+  ovalConfigs: OvalConfigs,
+): { valid: boolean; ovalHeaderConfigs: OvalHeaderConfigs | undefined } => {
+  if (!header) return { valid: true, ovalHeaderConfigs: undefined };
+  if (typeof header !== "string") return { valid: false, ovalHeaderConfigs: undefined };
+
+  let ovalHeaderConfigs: OvalHeaderConfigs;
+  let valid = true;
+  try {
+    ovalHeaderConfigs = JSON.parse(header);
+    if (!ovalHeaderConfigs.unlockAddresses)
+      throw new Error(`Value "${header}" cannot be converted to OvalConfigs records`);
+    if (!Array.isArray(ovalHeaderConfigs.unlockAddresses))
+      throw new Error(`Value "${header}" cannot be converted to OvalConfigs records`);
+    if (ovalHeaderConfigs.unlockAddresses.find((ovalHeaderConfig) => !isAddress(ovalHeaderConfig.ovalAddress)))
+      throw new Error(`Value "${header}" cannot be converted to OvalConfigs records`);
+    if (ovalHeaderConfigs.unlockAddresses.find((ovalHeaderConfig) => !ovalConfigs[ovalHeaderConfig.ovalAddress]))
+      throw new Error(`Value "${header}" cannot be converted to OvalConfigs records`);
+  } catch {
+    return { valid: false, ovalHeaderConfigs: undefined };
+  }
+  return { valid, ovalHeaderConfigs };
+};
+
+// Calculate refunds for the each unlock bundles this assumes the unlock bundles are the first elements in the bundle
+// and the unlockAddresses are in the same order as the unlock bundles.
+export const calculateBundleRefunds = (unlockAddresses: string[], ovalConfigs: OvalConfigs): Refund[] => {
+  const refunds = new Map<string, Refund>();
+  // We split the refund percentage equally between the unlock addresses and for each we calculate the refund according
+  // Oval instance refund percentage in OvalConfigs.
+  const split = 100 / unlockAddresses.length;
+  for (let i = 0; i < unlockAddresses.length; i++) {
+    const ovalAddress = unlockAddresses[i];
+    const refundToAdd = floorToDecimals((split * ovalConfigs[ovalAddress].refundPercent) / 100, 3);
```

Comment:
we only check `isAddress` in `getOvalConfigs`, so this would not work if address formatting differ (e.g. checksummed case in `ovalConfigs` while all lowercase in header.

As we now allow to pass oval addresses, we should normalize to all checksummed both in `getOvalConfigs` and `getOvalHeaderConfigs` so that we can safely match `ovalConfigs[ovalAddress]`